### ISSUE TITLE
shipit_pulse_listener: Retry failed code coverage tasks

### ISSUE THREAD

**Issue by marco-c** [mozilla-releng/services#704]
Most of the failures are due to transient issues, we should implement some kind of retry mechanism.

Does it apply to static analysis too, @La0 @jankeromnes ?

**Comment by mutterroland** [comment]
I would love to have some more infos. Maybe I can have a try on that one :).

**Comment by marco-c** [comment]
@mutterroland this issue requires knowledge of asyncio, are you familiar with it? If you aren't, I'd suggest working on another good-first-bug.

**Comment by mutterroland** [comment]
@marco-c I will move to another good-first-bug.

**Comment by jankeromnes** [comment]
> Does it apply to static analysis too?

Yes, it just happened during today's release, because pulse-listener and static-analysis didn't update at exactly the same time (in this window, an incompatibility between bot versions caused a few analyses to fail).

**Comment by La0** [comment]
The [retrigger code](https://github.com/mozilla/release-services/pull/1578) could be reused here.

**Comment by La0** [comment]
I'm starting to work on this one, here is what i can do:

1. Retry immediately tasks in exception state (with a maximum of N retries - N=3). Exception states are mostly due to Taskcluster (workers down, timeouts, ...)
2. Retry tasks in error state once, with a waiting time

If I use the retrigger code, a new task group & task id will be created, so I need to keep track of this in the code.
The [rerunTask](https://docs.taskcluster.net/docs/reference/platform/taskcluster-queue/references/api#rerunTask) would be much cleaner but is marked deprecated (i asked on #taskcluster about an alternative).

**Comment by La0** [comment]
From #taskcluster:

> <&pmoore> | bastien: although it isn't documented there, i believe the preferred approach is to create a new task with the same task definition, but with a different taskId. we should probably update the docs
> <&pmoore> | bastien: i've created https://github.com/taskcluster/taskcluster-queue/pull/292 for this (feedback welcome as this is a particularly delicate issue - aki-away/bstack/dustin/jhford)

**Comment by La0** [comment]
Bumping this, as it's blocking static analysis users: #1846
